lib/block_view.go

```diff
@@ -7009,6 +8537,43 @@ func (bav *UtxoView) GetUnspentUtxoEntrysForPublicKey(pkBytes []byte) ([]*UtxoEn
 	return utxoEntriesToReturn, nil
 }
 
+func (bav *UtxoView) GetSpendableBitcloutBalanceNanosForPublicKey(pkBytes []byte,
+) (_spendableBalance uint64, _err error) {
+	// In order to get the spendable balance, we need to account for any immature block rewards.
+	// We get these by starting at the chain tip and iterating backwards until we have collected
+	// all of the immature block rewards for this public key.
+	nextBlockHash := bav.TipHash
+	numImmatureBlocks := uint64(bav.Params.BlockRewardMaturity / bav.Params.TimeBetweenBlocks)
+	immatureBlockRewards := uint64(0)
+	for ii := uint64(1); ii < numImmatureBlocks; ii++ {
+		// Don't look up the genesis block since it isn't in the DB.
+		if GenesisBlockHashHex == nextBlockHash.String() {
+			break
+		}
+
+		block, err := GetBlock(nextBlockHash, bav.Handle)
```

Comment:
Oof.. kindof annoying that we have to do a GetBlock call here since that's a pretty heavy-weight thing. But I guess this is just another instance where Postgres will save us...